packages/palette/src/utils/useInterval.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef } from "react"
+
+/** useInterval */
+export function useInterval(callback, delay) {
+  const savedCallback = useRef(null)
+
+  // Remember the latest callback.
+  useEffect(() => {
```

Comment:
The code in this hook isn't totally obvious to me. Are there any tests we could put around this hook that could describe how these hooks are all intended to work together? Nothing that tests the exact implementation directly, but something that renders a test component that uses `useInterval`, and gets a callback fired the expected number of times within a window of time.